docs: updating tutorial for ismb (#469)
Reed-CompBio/spras: docs/tutorial/intermediate.rst

 Step 1: Transforming high throughput experimental data into SPRAS compatible input data
*****************************************************************************************

> 1.1 Example of high throughput omic data

Reviewer: ```suggestion
1.1 Example of high-throughput omic data
```

points, generating large-scale datasets that can be used as input for
pathway reconstruction.

An example dataset is EGF response mass spectrometry data [4]_, a
proteomics dataset that measures peptide abundance after cells are
stimulated with epidermal growth factor (EGF).

The experiment for this data was repeated three times, known as
biological replicates, to ensure the results are consistent. Each
replicate measures the abundance of peptides at different time points
(0-128 minutes) to capture how protein activity changes over time.

.. note::

   Mass spectrometry is a technique used to measure and identify
   proteins in a sample. It works by breaking proteins into smaller
   pieces called peptides and measuring their mass-to-charge ratio,
   which enables identifying which peptide is being measured. The data
   show how much of each peptide is present, which can show how protein
   phosphorylation abundances change under different conditions.

   Since proteins interact with each other in biological pathways,
   changes in their phosphorylation abundances can reveal which parts of
   a pathway are active or affected.

> Example of one of the biological replicate A with one peptide:

Reviewer: ```suggestion
Example of one peptide's measurements in one of the biological replicates:
```

Similar automated steps from the previous tutorial runs behind the
scenes for ``intermediate.yaml``. However, this configuration now runs
multiple algorithms with different parameter combinations, which takes

> longer to complete. By increasing the number of cores to 8, it allows

Reviewer: Does the codespace support this? My choices were 2 or 4.

Author: I forgot to change this back to 4.

``output/intermediate/egfr-ml/hac-horizontal.png`` and/or
``output/intermediate/egfr-ml/hac-vertical.png`` and open it locally.

> SPRAS includes hierarchical agglomerative clustering (HAC) to group

Reviewer: Can we avoid the HAC algorithm? I don't think it is that common.

.. note::

   The gold standard for this dataset consists of nodes only, following
   the original study. The gold standard nodes are drawn from eight
   EGFR-related reference pathways [4]_.

   A limitation of this gold standard is its incomplete coverage of EGF
   signaling pathways. Across the eight reference pathways, typically
   5\% or fewer of the input nodes appear in any single pathway, and
   85\% are absent from all eight. This reflects the general
   incompleteness of curated pathway databases relative to measured
   signaling responses, rather than a flaw specific to this dataset
   [4]_.

With these updates, SPRAS will run the evaluations across all outputs
for a given dataset.

After saving the changes in the configuration file, rerun with:

.. code:: bash

   snakemake --cores 4 --configfile config/intermediate.yaml

What happens when you run this command
--------------------------------------

#. Reusing cached results

Snakemake reads the options set in ``intermediate.yaml`` and checks for
any requested post-analysis steps. It reuses cached results; here the
``pathway.txt`` files generated from the previously executed algorithms
on the egfr dataset are reused.

> 2. Running the ml analysis

Reviewer: I typically see "ml" capitalized "ML"  and would do that in text (i.e. not the config file). Did we define that acronym?

Author: I'll double check to make sure I did define it.
